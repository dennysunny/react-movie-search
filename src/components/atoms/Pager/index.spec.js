import React from 'react';
import { shallow } from 'enzyme';

import MyPager from '.';

const mockProps = {
  page: 1,
  handlePageSelect: jest.fn(),
};

describe('Atoms/MyPager', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<MyPager {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle onSelect event', () => {
    const wrapper = shallow(<MyPager {...mockProps} />);

    wrapper.find(Pager.Item).first().simulate('click');
    expect(mockProps.handlePageSelect).toHaveBeenCalled();
  });
});
