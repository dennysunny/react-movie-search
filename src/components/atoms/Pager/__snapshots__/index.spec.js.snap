// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/MyPager should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyPager
    handlePageSelect={[MockFunction]}
    page={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "pager",
      "children": Array [
        <PagerItem
          disabled={false}
          eventKey={0}
          href="#"
          next={false}
          onClick={[MockFunction]}
          previous={true}
        >
          ← Previous
        </PagerItem>,
        <PagerItem
          disabled={false}
          eventKey={2}
          next={true}
          onClick={[MockFunction]}
          previous={false}
        >
          Next →
        </PagerItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "← Previous",
          "disabled": false,
          "eventKey": 0,
          "href": "#",
          "next": false,
          "onClick": [MockFunction],
          "previous": true,
        },
        "ref": null,
        "rendered": "← Previous",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Next →",
          "disabled": false,
          "eventKey": 2,
          "next": true,
          "onClick": [MockFunction],
          "previous": false,
        },
        "ref": null,
        "rendered": "Next →",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "pager",
        "children": Array [
          <PagerItem
            disabled={false}
            eventKey={0}
            href="#"
            next={false}
            onClick={[MockFunction]}
            previous={true}
          >
            ← Previous
          </PagerItem>,
          <PagerItem
            disabled={false}
            eventKey={2}
            next={true}
            onClick={[MockFunction]}
            previous={false}
          >
            Next →
          </PagerItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "← Previous",
            "disabled": false,
            "eventKey": 0,
            "href": "#",
            "next": false,
            "onClick": [MockFunction],
            "previous": true,
          },
          "ref": null,
          "rendered": "← Previous",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Next →",
            "disabled": false,
            "eventKey": 2,
            "next": true,
            "onClick": [MockFunction],
            "previous": false,
          },
          "ref": null,
          "rendered": "Next →",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
